internal/pkg/image/driver/imagedriver.go

```diff
@@ -307,20 +320,24 @@ func (d *fuseappsDriver) Mount(params *image.MountParams, mfunc image.MountFunc)
 		time.Sleep(sleepTime)
 		totTime += sleepTime
 		var ws syscall.WaitStatus
-		wpid, err := syscall.Wait4(process.Pid, &ws, syscall.WNOHANG, nil)
-		if err != nil {
-			return fmt.Errorf("unable to get wait status on %v: %v: %v", f.binName, err, filterMsg())
-		}
-		if wpid != 0 {
-			msg := filterMsg()
-			if strings.Contains(msg, "fusermount") {
-				sylog.Infof("A fusermount error indicates that the kernel is too old")
-				if params.Filesystem == "squashfs" {
-					sylog.Infof("The --unsquash option may work around it")
+		// gocryptfs has a separated notifypid
```

Comment:
Why does it need a notifypid?

Using "separate" in this comment would be better than "separated".

Reply:
gocryptfs is spawned in a separate process
resolved in this commit https://github.com/apptainer/apptainer/pull/1185/commits/9c7ebbd0175ce54089dd7e259bc6b6ef98584c1a

Reply:
unlike squashfs, when we start gocryptfs via `exec.Command`, the gocryptfs process cmdline is like this
```
vagrant   879765       1  0 Mar30 ?        00:00:00 /usr/local/bin/gocryptfs -fg -notifypid=879759 /tmp/bundle-temp-680496664/gocryptfs-3985556534/cipher /tmp/bundle-temp-680496664/gocryptfs-3985556534/plain
```
where `-notifypid=879759`, the `879759` is the value of `cmd.Process.Pid`, and gocryptfs itself is running with another pid (879765). So we can not directly use the `cmd.Process.Pid` but have to search for its real pid via `-notifypid=xxx`

Reply:
Can't you just invoke gocryptfs with `--fg`?  That corresponds with the `-f` option that is used for the other 3 driver types, to run in the foreground.  I tried that and I don't see it using `--notifypid` then.

Reply:
yeah, we can set `-fg` to gocryptfs to explicitly ask it to stay foreground. I have refactored the code to implement it by putting the code inside a separate goroutine.